tests/test_properties.py

```diff
@@ -30,6 +31,24 @@ def test_roundtrip(data: st.DataObject, zarr_format: int) -> None:
     assert_array_equal(nparray, zarray[:])
 
 
+@given(array=arrays())
+def test_array_creates_implicit_groups(array):
+    path = array.path
+    ancestry = path.split("/")[:-1]
+    for i in range(len(ancestry)):
+        parent = "/".join(ancestry[: i + 1])
+        if array.metadata.zarr_format == 2:
+            assert (
+                sync(array.store.get(f"{parent}/.zgroup", prototype=default_buffer_prototype()))
+                is not None
+            )
+        elif array.metadata.zarr_format == 3:
+            assert (
+                sync(array.store.get(f"{parent}/zarr.json", prototype=default_buffer_prototype()))
+                is not None
+            )
```

Comment:
looks good, thanks for adding this. #2665 will include a function that makes retrieving an array or group from a path in storage that we can use instead of calling `store.get` here.